code/datums/status_effects/debuffs/blindness.dm

```diff
@@ -68,11 +69,21 @@
 	if(!CAN_BE_BLIND(owner))
 		return FALSE
 
+	if(HAS_TRAIT(owner, TRAIT_SIGHT_BYPASS))
+		RegisterSignal(owner, SIGNAL_REMOVETRAIT(TRAIT_SIGHT_BYPASS), PROC_REF(on_apply))
```

Comment:
1. You would need to make this signal point to another proc which calls us because `on_apply` is not a signal handler and you can't make it one
2. I would prefer if you separated out all "make the guy blind" code into a separate proc, and had this signal + on apply call that proc, instead of calling `on_apply` directly. It's a core status effect proc, this might break stuff in the future
```dm
/datum/status_effect/grouped/blindness/on_apply()
	make_blind()

/datum/status_effect/grouped/blindness/proc/make_blind()
	RegisterSignal(owner, SIGNAL_ADDTRAIT(TRAIT_SIGHT_BYPASS), PROC_REF(temporary_clear))
	owner.overlay_fullscreen(id, /atom/movable/screen/fullscreen/blind)
	// You are blind - at most, able to make out shapes near you
	owner.add_client_colour(/datum/client_colour/monochrome/blind)
```

Reply:
i tihnk i fixed it